src/drivers/github.go

```diff
@@ -8,21 +8,46 @@ import (
 
 	"golang.org/x/oauth2"
 
-	"github.com/git-town/git-town/src/git"
 	"github.com/google/go-github/github"
 )
 
+type githubConfig interface {
+	GetCodeHostingDriverName() string
+	GetMainBranch() string
+	GetURLRepositoryName(string) string
+	GetURLHostname(string) string
+	GetCodeHostingOriginHostname() string
+	GetRemoteOriginURL() string
+	GetGitHubToken() string
+}
+
 type githubCodeHostingDriver struct {
 	originURL  string
 	hostname   string
 	apiToken   string
 	client     *github.Client
 	owner      string
 	repository string
+	mainBranch string
 }
 
-func (d *githubCodeHostingDriver) CanBeUsed(driverType string) bool {
-	return driverType == "github" || d.hostname == "github.com"
+func tryCreateGitHub(config githubConfig) CodeHostingDriver {
+	if config.GetCodeHostingDriverName() != "github" && config.GetCodeHostingOriginHostname() != "github.com" {
+		return nil
+	}
+	originURL := config.GetRemoteOriginURL()
+	result := githubCodeHostingDriver{
+		originURL:  originURL,
+		hostname:   config.GetURLHostname(originURL),
+		apiToken:   config.GetGitHubToken(),
+		mainBranch: config.GetMainBranch(),
```

Comment:
Is using the inferface?

Reply:
Yes, `config` is of type `githubConfig` here, which is an interface.